data_prep.py

```diff
@@ -1,150 +1,330 @@
-import matplotlib.pyplot as plt
+import os
+from datetime import datetime
+from pathlib import Path
+
 import numpy as np
 import pandas as pd
+from loguru import logger
+
+
+class Pipeline:
+    def __init__(
+        self,
+        data_path: str = Path(os.getcwd()) / "WEC2022_Data",
```

Comment:
I think it is better to use
`Path(__file__)`,  then 
`Path(os.getcwd()) or Path(os.path.abspath(""))`
since __file__ is not dependend of directory where the main script was started.
For example when main.py is in root directory, `Path(os.getcwd())` will be root directory. 
On contrary, `Path(__file__)` will stay the same, since it is attached to certain file

Reply:
Yet, when using`Path(__file__)`, it gives file path, not directory path. In order to get directory path, You can do `Path(__file__).parents[0]`

Reply:
used Path(__file__) / ".." to solve the issue